pkg/ig-manager/ig-manager.go

```diff
@@ -83,6 +85,11 @@ func NewManager(runtimes []*containerutils.RuntimeConfig) (*IGManager, error) {
 		return nil, err
 	}
 
+	// Check this now otherwise below RemoveMemlock will fail.
+	if os.Geteuid() != 0 {
+		return nil, errors.New("ig must be run as root to be able to run eBPF programs")
```

Comment:
I tested it and it looks like this:
```bash
$ ./ig trace exec
Error: initializing operators: initializing operator "LocalManager": initializing manager: ig must be run as root to be able to run eBPF programs
```

Before this patch, it looked like this:
```bash
$ ./ig trace exec
Error: ./ig must be run as root to be able to run eBPF programs
```

I think the more concise output is better.

Reply:
Indeed, nothing is perfect. I think your suggestion 3 is the best.

```go
import  "golang.org/x/exp/slices"

verbsAllowedByNonRoot := []string{"help", "completion", "version"}
if os.Geteuid() != 0 &&  len(os.args) > 1 && !slices.Contains(verbsAllowedByNonRoot, os.args[1]) {
}
```

Reply:
What about adding the check in [here](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/runtime/local/local.go#L53) - that is always called before running an actual gadget and in the error case it should return `Error: initializing runtime: ig must be run as root to be able to run eBPF programs`. I think that could be a good compromise.

Reply:
> I just remember about the existing of os.Args[0] for this case :p!

You might want to wrap that in a `filepath.Base(os.Args[0])`, then - otherwise, if called with a path, that would show up as well.